pkg/cmd/start/main.go

```diff
@@ -191,6 +195,68 @@ func availableAPIs(kubeconfig *rest.Config) (*metav1.APIGroupList, error) {
 	return apiList, nil
 }
 
+func detectPlatform(apiList *metav1.APIGroupList) {
+	// detect the platform
+	if strings.EqualFold(viper.GetString("platform"), v1.FlagPlatformAutoDetect) {
+		log.Debug("Attempting to auto-detect the platform")
+		if isOpenShift(apiList) {
+			viper.Set("platform", v1.FlagPlatformOpenShift)
+		} else {
+			viper.Set("platform", v1.FlagPlatformKubernetes)
+		}
+
+		log.WithField("platform", viper.GetString("platform")).Info("Auto-detected the platform")
+	} else {
+		log.WithField("platform", viper.GetString("platform")).Debug("The 'platform' option is explicitly set")
+	}
+}
+
+func detectElasticsearch(apiList *metav1.APIGroupList) {
+	// detect whether the Elasticsearch operator is available
+	if strings.EqualFold(viper.GetString("es-provision"), v1.FlagProvisionElasticsearchAuto) {
+		log.Debug("Determining whether we should enable the Elasticsearch Operator integration")
+		if isElasticsearchOperatorAvailable(apiList) {
+			viper.Set("es-provision", v1.FlagProvisionElasticsearchTrue)
+		} else {
+			viper.Set("es-provision", v1.FlagProvisionElasticsearchFalse)
+		}
+
+		log.WithField("es-provision", viper.GetString("es-provision")).Info("Automatically adjusted the 'es-provision' flag")
+	} else {
+		log.WithField("es-provision", viper.GetString("es-provision")).Debug("The 'es-provision' option is explicitly set")
+	}
+}
+
+func detectClusterRoles(mgr manager.Manager) {
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
```

Comment:
This deserves an explanation: I faced operator-framework/operator-sdk#1667 during this, and it looks like it might be because the cache isn't started. If I can find the proper way soon, I'll amend this PR, otherwise, I'll create a follow-up ticket to get this to use the same client as the other parts.